app/src/main/java/com/chouten/app/ui/views/search/SearchPageViewModel.kt

```diff
@@ -86,76 +150,15 @@ class SearchPageViewModel(
-
-            searchFns?.forEach { searchFn ->
-                // We need the search function to
-                // be executed synchronously.
-                viewModelScope.launch {
-                    if (!webview.load(
-                            searchFn.copy(
-                                request = searchFn.request?.url?.let {
-                                    searchFn.request.copy(
-                                        url = it.replace(
-                                            "<query>",
-                                            withContext(Dispatchers.IO) {
-                                                URLEncoder.encode(query, "UTF-8")
-                                            }
-                                        )
-                                    )
-                                }
-                            )
-                        )
-                    ) {
-                        isSearching = false
-                        return@launch
-                    }
 
-                    val res = webview.inject(searchFn)
-                    if (res.isBlank()) {
-                        PrimaryDataLayer.enqueueSnackbar(
-                            SnackbarVisualsWithError(
-                                "No results found for $query",
-                                false
-                            )
-                        )
-                        isSearching = false
-                        return@launch
-                    }
+        val code = this.getCode();
 
-                    _searchResults.clear()
-                    try {
-                        val results =
-                            Mapper.parse<ModuleResponse<List<SearchResult>>>(
-                                res
-                            )
-                        _searchResults.addAll(results.result)
-                        isSearching = false
-                        webview.updateNextUrl(results.nextUrl)
-                    } catch (e: Exception) {
-                        PrimaryDataLayer.enqueueSnackbar(
-                            SnackbarVisualsWithError(
-                                "Error parsing search results for $query", true
-                            )
-                        )
-                        e.printStackTrace()
-                        e.localizedMessage?.let {
-                            LogEntry(
-                                title = "Error parsing search results for $query",
-                                message = it,
-                                isError = true,
-                            )
-                        }?.let {
-                            LogLayer.addLogEntry(
-                                it
-                            )
-                        }
-                        isSearching = false
-                    }
-                }
-            }
+        if(!code.isEmpty()){
+            val webviewPayload = mapOf<String, String>(
```

Comment:
Maybe consider using a class so that you can leverage the `Mapper`? It'd prevent possible issues wrt key misspellings

Reply:
TODO added.